packages/lesswrong/lib/collections/posts/views.ts

```diff
@@ -868,22 +869,22 @@ Posts.addView("onlineEvents", (terms: PostsViewTerms) => {
   return query
 })
 ensureIndex(Posts,
-  augmentForDefaultView({ onlineEvent:1, startTime:1 }),
-  { name: "posts.onlineEvents" }
+  augmentForDefaultView({ globalEvent:1, startTime:1 }),
+  { name: "posts.globalEvents" }
 );
 
 Posts.addView("nearbyEvents", (terms: PostsViewTerms) => {
   const timeSelector = {$or: [
     {startTime: {$gt: moment().subtract(eventBuffer.startBuffer).toDate()}},
     {endTime: {$gt: moment().subtract(eventBuffer.endBuffer).toDate()}}
   ]}
-  const onlineEvent = terms.onlineEvent === false ? false : viewFieldAllowAny
+
   let query: any = {
     selector: {
       location: {$exists: true},
       groupId: null,
       isEvent: true,
-      onlineEvent: onlineEvent,
+      ...(terms.globalEvent === undefined ? {} : {globalEvent: terms.globalEvent}),
```

Comment:
This has schemaDefaultValue(false), so you want to test for falseyness I believe. (Old documents will be undefined, new ones will be false, I think.)

Reply:
Thanks for flagging this - it took me a while to figure out how to fix it because I didn't realize my new `$or` for the `globalEvent` field was getting overwritten by the `$or` from `timeSelector`, turns out I needed to put those inside of an `$and`.

Reply:
Ouch, yeah that sucks.